pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -475,6 +477,7 @@ private void writeMetadata()
     properties.setProperty(DIMENSIONS, _config.getDimensions());
     properties.setProperty(METRICS, _config.getMetrics());
     properties.setProperty(DATETIME_COLUMNS, _config.getDateTimeColumnNames());
+    properties.setProperty(COMPLEX_COLUMNS, _config.getComplexColumnNames());
```

Comment:
I feel this is the reason for the crc change. Should we consider not adding this field if there is no complex column?

Reply:
Hmm, why did the test pass then

Reply:
It's not persisting if value is empty.